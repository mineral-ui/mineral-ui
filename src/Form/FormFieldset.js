/**
 * Copyright 2017 CA
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/* @flow */
import React from 'react';
import { createStyledComponent } from '../styles';

type Props = {
  /** Rendered content of the component, most likely [FormField](../form-field) */
  children?: React$Node,
  /** Disable (visually) the fieldset */
  disabled?: boolean,
  /** Title of the fieldset */
  legend?: string | React$Element<*>
};

export const componentTheme = (baseTheme: Object) => ({
  FormFieldset_borderColor: baseTheme.borderColor,

  FormFieldsetLegend_color_text: baseTheme.color_text,
  FormFieldsetLegend_fontSize: baseTheme.fontSize_ui,
  FormFieldsetLegend_fontWeight: baseTheme.fontWeight_bold,

  ...baseTheme
});

const Root = createStyledComponent(
  'fieldset',
  props => {
    const { disabled, theme: baseTheme } = props;
    const theme = componentTheme(baseTheme);

    return {
      border: '1px solid transparent',
      borderTopColor: theme.FormFieldset_borderColor,
      padding: 0,

      '& > legend': {
        color: disabled
          ? theme.color_text_disabled
          : theme.FormFieldsetLegend_color_text,
        fontSize: theme.FormFieldsetLegend_fontSize,
        fontWeight: theme.FormFieldsetLegend_fontWeight,
        lineHeight: theme.size_medium,
        padding: 0,
        paddingLeft: theme.direction === 'rtl' ? theme.space_inline_sm : null,
        paddingRight: theme.direction === 'ltr' ? theme.space_inline_sm : null
      }
    };
  },
  {
    displayName: 'FormFieldset',
    includeStyleReset: true,
    rootEl: 'fieldset'
  }
);

/**
 * FormFieldset wraps related [FormFields](../form-field) and provides a legend.
 */
export default function FormFieldset({
  children,
  legend,
  ...restProps
}: Props) {
  return (
    <Root {...restProps}>
      {legend && <legend>{legend}</legend>}
      {children}
    </Root>
  );
}
