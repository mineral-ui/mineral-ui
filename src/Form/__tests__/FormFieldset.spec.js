/* @flow */
import React from 'react';
import { shallow } from 'enzyme';
import FormFieldset from '../FormFieldset';
import examples from '../../website/app/demos/Form/examples/FormFieldset';
import testDemoExamples from '../../../utils/testDemoExamples';

function shallowFormFieldset(props = {}) {
  const formFieldsetProps = {
    ...props
  };

  return shallow(<FormFieldset {...formFieldsetProps} />);
}

describe('FormFieldset', () => {
  describe('renders', () => {
    it('root', () => {
      const formFieldset = shallowFormFieldset();

      expect(formFieldset.exists()).toEqual(true);
    });
  });

  testDemoExamples(examples);
});
