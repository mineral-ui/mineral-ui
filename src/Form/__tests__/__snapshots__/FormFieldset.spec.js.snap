// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldset demo examples basic 1`] = `
.glamor-9,
[data-glamor-9] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  border: 1px solid transparent;
  border-top-color: #c8d1e0;
  padding: 0;
}

.glamor-9 *,
[data-glamor-9] *,
.glamor-9 *::before,
[data-glamor-9] *::before,
.glamor-9 *::after,
[data-glamor-9] *::after {
  box-sizing: inherit;
}

.glamor-9 > legend,
[data-glamor-9] > legend {
  color: #333840;
  font-size: 0.875em;
  font-weight: 700;
  line-height: 2em;
  padding: 0;
  padding-right: 0.5em;
}

.glamor-3,
[data-glamor-3] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
}

.glamor-3 *,
[data-glamor-3] *,
.glamor-3 *::before,
[data-glamor-3] *::before,
.glamor-3 *::after,
[data-glamor-3] *::after {
  box-sizing: inherit;
}

.glamor-0,
[data-glamor-0] {
  color: #58606e;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  font-size: 0.875em;
  font-weight: 600;
  justify-content: space-between;
  margin-bottom: 0.5em;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0 > *,
[data-glamor-0] > * {
  align-self: flex-end;
  display: inline-block;
  -webkit-align-self: flex-end;
}

.glamor-2,
[data-glamor-2] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  align-items: center;
  cursor: text;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  position: relative;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-2 *,
[data-glamor-2] *,
.glamor-2 *::before,
[data-glamor-2] *::before,
.glamor-2 *::after,
[data-glamor-2] *::after {
  box-sizing: inherit;
}

.glamor-2 [role="img"],
[data-glamor-2] [role="img"] {
  display: block;
  fill: #c8d1e0;
  flex: 0 0 auto;
  margin: 0 0.5em;
  -webkit-flex: 0 0 auto;
}

.glamor-2 [role="img"]:last-of-type,
[data-glamor-2] [role="img"]:last-of-type {
  fill: #c8d1e0;
}

.glamor-1,
[data-glamor-1] {
  background-color: transparent;
  border: 0;
  box-shadow: none;
  color: #58606e;
  flex: 1 1 auto;
  font-family: inherit;
  font-size: 0.875em;
  height: 2.857142857142857em;
  min-width: 0;
  outline: 0;
  padding-left: 1.1428571428571428em;
  padding-right: 1.1428571428571428em;
  width: 100%;
  -webkit-flex: 1 1 auto;
}

.glamor-1::placeholder,
[data-glamor-1]::placeholder,
.glamor-1[data-simulate-placeholder],
[data-glamor-1][data-simulate-placeholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1::-ms-input-placeholder,
[data-glamor-1]::-ms-input-placeholder,
.glamor-1[data-simulate-msinputplaceholder],
[data-glamor-1][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1:-ms-input-placeholder,
[data-glamor-1]:-ms-input-placeholder,
.glamor-1[data-simulate-msinputplaceholder],
[data-glamor-1][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1::-ms-clear,
[data-glamor-1]::-ms-clear,
.glamor-1[data-simulate-msclear],
[data-glamor-1][data-simulate-msclear] {
  display: none;
}

.glamor-1:hover ~ div,
[data-glamor-1]:hover ~ div,
.glamor-1[data-simulate-hover] ~ div,
[data-glamor-1][data-simulate-hover] ~ div {
  border-color: #2e6fd9;
}

.glamor-1:focus ~ div,
[data-glamor-1]:focus ~ div,
.glamor-1[data-simulate-focus] ~ div,
[data-glamor-1][data-simulate-focus] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-1:active ~ div,
[data-glamor-1]:active ~ div,
.glamor-1[data-simulate-active] ~ div,
[data-glamor-1][data-simulate-active] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-1 ~ div,
[data-glamor-1] ~ div {
  background-color: #fff;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

<ThemeProvider>
  <ThemeProvider>
    <LiveProvider>
      <div
        className="react-live"
        scope={
          Object {
            "DemoLayout": [Function],
            "FormField": [Function],
            "FormFieldset": [Function],
            "TextInput": [Function],
          }
        }
      >
        <LivePreview>
          <div
            className="react-live-preview"
          >
            <FormFieldset
              legend="Login"
            >
              <FormFieldset>
                <fieldset
                  className="glamor-9"
                >
                  <legend>
                    Login
                  </legend>
                  <glamorous(div)>
                    <div
                      className="glamor-8"
                    >
                      <FormField
                        label="Email"
                        requiredText="Required"
                      >
                        <FormField
                          className={undefined}
                        >
                          <div
                            className="glamor-3"
                          >
                            <label>
                              <glamorous(div)
                                hideLabel={undefined}
                                key="formField-1-textWrapper"
                              >
                                <div
                                  className="glamor-0"
                                >
                                  <span
                                    id="formField-1-labelText"
                                  >
                                    Email
                                  </span>
                                </div>
                              </glamorous(div)>
                              <TextInput
                                aria-describedby={undefined}
                                key="formField-1-control"
                                required={undefined}
                                rootProps={undefined}
                                type="email"
                                variant={undefined}
                              >
                                <TextInput
                                  className={undefined}
                                  variant={undefined}
                                >
                                  <div
                                    className="glamor-2"
                                  >
                                    <glamorous(input)
                                      aria-describedby={undefined}
                                      aria-invalid={undefined}
                                      aria-required={undefined}
                                      disabled={undefined}
                                      iconEnd={undefined}
                                      iconStart={undefined}
                                      innerRef={[Function]}
                                      prefix={undefined}
                                      readOnly={undefined}
                                      required={undefined}
                                      size="large"
                                      suffix={undefined}
                                      type="email"
                                      variant={undefined}
                                    >
                                      <input
                                        aria-describedby={undefined}
                                        aria-invalid={undefined}
                                        aria-required={undefined}
                                        className="glamor-1"
                                        disabled={undefined}
                                        prefix={undefined}
                                        readOnly={undefined}
                                        required={undefined}
                                        size="large"
                                        type="email"
                                      />
                                    </glamorous(input)>
                                    <div />
                                  </div>
                                </TextInput>
                              </TextInput>
                            </label>
                          </div>
                        </FormField>
                      </FormField>
                      <FormField
                        label="Password"
                        requiredText="Required"
                      >
                        <FormField
                          className={undefined}
                        >
                          <div
                            className="glamor-3"
                          >
                            <label>
                              <glamorous(div)
                                hideLabel={undefined}
                                key="formField-2-textWrapper"
                              >
                                <div
                                  className="glamor-0"
                                >
                                  <span
                                    id="formField-2-labelText"
                                  >
                                    Password
                                  </span>
                                </div>
                              </glamorous(div)>
                              <TextInput
                                aria-describedby={undefined}
                                key="formField-2-control"
                                required={undefined}
                                rootProps={undefined}
                                type="password"
                                variant={undefined}
                              >
                                <TextInput
                                  className={undefined}
                                  variant={undefined}
                                >
                                  <div
                                    className="glamor-2"
                                  >
                                    <glamorous(input)
                                      aria-describedby={undefined}
                                      aria-invalid={undefined}
                                      aria-required={undefined}
                                      disabled={undefined}
                                      iconEnd={undefined}
                                      iconStart={undefined}
                                      innerRef={[Function]}
                                      prefix={undefined}
                                      readOnly={undefined}
                                      required={undefined}
                                      size="large"
                                      suffix={undefined}
                                      type="password"
                                      variant={undefined}
                                    >
                                      <input
                                        aria-describedby={undefined}
                                        aria-invalid={undefined}
                                        aria-required={undefined}
                                        className="glamor-1"
                                        disabled={undefined}
                                        prefix={undefined}
                                        readOnly={undefined}
                                        required={undefined}
                                        size="large"
                                        type="password"
                                      />
                                    </glamorous(input)>
                                    <div />
                                  </div>
                                </TextInput>
                              </TextInput>
                            </label>
                          </div>
                        </FormField>
                      </FormField>
                    </div>
                  </glamorous(div)>
                </fieldset>
              </FormFieldset>
            </FormFieldset>
          </div>
        </LivePreview>
      </div>
    </LiveProvider>
  </ThemeProvider>
</ThemeProvider>
`;

exports[`FormFieldset demo examples disabled 1`] = `
.glamor-3,
[data-glamor-3] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
}

.glamor-3 *,
[data-glamor-3] *,
.glamor-3 *::before,
[data-glamor-3] *::before,
.glamor-3 *::after,
[data-glamor-3] *::after {
  box-sizing: inherit;
}

.glamor-0,
[data-glamor-0] {
  color: #58606e;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  font-size: 0.875em;
  font-weight: 600;
  justify-content: space-between;
  margin-bottom: 0.5em;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0 > *,
[data-glamor-0] > * {
  align-self: flex-end;
  display: inline-block;
  -webkit-align-self: flex-end;
}

.glamor-2,
[data-glamor-2] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  align-items: center;
  cursor: text;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  position: relative;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-2 *,
[data-glamor-2] *,
.glamor-2 *::before,
[data-glamor-2] *::before,
.glamor-2 *::after,
[data-glamor-2] *::after {
  box-sizing: inherit;
}

.glamor-2 [role="img"],
[data-glamor-2] [role="img"] {
  display: block;
  fill: #c8d1e0;
  flex: 0 0 auto;
  margin: 0 0.5em;
  -webkit-flex: 0 0 auto;
}

.glamor-2 [role="img"]:last-of-type,
[data-glamor-2] [role="img"]:last-of-type {
  fill: #c8d1e0;
}

.glamor-9,
[data-glamor-9] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  border: 1px solid transparent;
  border-top-color: #c8d1e0;
  padding: 0;
}

.glamor-9 *,
[data-glamor-9] *,
.glamor-9 *::before,
[data-glamor-9] *::before,
.glamor-9 *::after,
[data-glamor-9] *::after {
  box-sizing: inherit;
}

.glamor-9 > legend,
[data-glamor-9] > legend {
  color: #afbacc;
  font-size: 0.875em;
  font-weight: 700;
  line-height: 2em;
  padding: 0;
  padding-right: 0.5em;
}

.glamor-1,
[data-glamor-1] {
  background-color: transparent;
  border: 0;
  box-shadow: none;
  color: #afbacc;
  flex: 1 1 auto;
  font-family: inherit;
  font-size: 0.875em;
  height: 2.857142857142857em;
  min-width: 0;
  outline: 0;
  padding-left: 1.1428571428571428em;
  padding-right: 1.1428571428571428em;
  width: 100%;
  -webkit-flex: 1 1 auto;
}

.glamor-1::placeholder,
[data-glamor-1]::placeholder,
.glamor-1[data-simulate-placeholder],
[data-glamor-1][data-simulate-placeholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1::-ms-input-placeholder,
[data-glamor-1]::-ms-input-placeholder,
.glamor-1[data-simulate-msinputplaceholder],
[data-glamor-1][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1:-ms-input-placeholder,
[data-glamor-1]:-ms-input-placeholder,
.glamor-1[data-simulate-msinputplaceholder],
[data-glamor-1][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-1::-ms-clear,
[data-glamor-1]::-ms-clear,
.glamor-1[data-simulate-msclear],
[data-glamor-1][data-simulate-msclear] {
  display: none;
}

.glamor-1:focus ~ div,
[data-glamor-1]:focus ~ div,
.glamor-1[data-simulate-focus] ~ div,
[data-glamor-1][data-simulate-focus] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-1:active ~ div,
[data-glamor-1]:active ~ div,
.glamor-1[data-simulate-active] ~ div,
[data-glamor-1][data-simulate-active] ~ div {
  border-color: #c8d1e0;
  box-shadow: none;
}

.glamor-1 ~ div,
[data-glamor-1] ~ div {
  background-color: #ebeff5;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

<ThemeProvider>
  <ThemeProvider>
    <LiveProvider>
      <div
        className="react-live"
        scope={
          Object {
            "DemoLayout": [Function],
            "FormField": [Function],
            "FormFieldset": [Function],
            "TextInput": [Function],
          }
        }
      >
        <LivePreview>
          <div
            className="react-live-preview"
          >
            <FormFieldset
              disabled={true}
              legend="Login"
            >
              <FormFieldset
                disabled={true}
              >
                <fieldset
                  className="glamor-9"
                  disabled={true}
                >
                  <legend>
                    Login
                  </legend>
                  <glamorous(div)>
                    <div
                      className="glamor-8"
                    >
                      <FormField
                        label="Email"
                        requiredText="Required"
                      >
                        <FormField
                          className={undefined}
                        >
                          <div
                            className="glamor-3"
                          >
                            <label>
                              <glamorous(div)
                                hideLabel={undefined}
                                key="formField-3-textWrapper"
                              >
                                <div
                                  className="glamor-0"
                                >
                                  <span
                                    id="formField-3-labelText"
                                  >
                                    Email
                                  </span>
                                </div>
                              </glamorous(div)>
                              <TextInput
                                aria-describedby={undefined}
                                disabled={true}
                                key="formField-3-control"
                                required={undefined}
                                rootProps={undefined}
                                type="email"
                                variant={undefined}
                              >
                                <TextInput
                                  className={undefined}
                                  variant={undefined}
                                >
                                  <div
                                    className="glamor-2"
                                  >
                                    <glamorous(input)
                                      aria-describedby={undefined}
                                      aria-invalid={undefined}
                                      aria-required={undefined}
                                      disabled={true}
                                      iconEnd={undefined}
                                      iconStart={undefined}
                                      innerRef={[Function]}
                                      prefix={undefined}
                                      readOnly={undefined}
                                      required={undefined}
                                      size="large"
                                      suffix={undefined}
                                      type="email"
                                      variant={undefined}
                                    >
                                      <input
                                        aria-describedby={undefined}
                                        aria-invalid={undefined}
                                        aria-required={undefined}
                                        className="glamor-1"
                                        disabled={true}
                                        prefix={undefined}
                                        readOnly={undefined}
                                        required={undefined}
                                        size="large"
                                        type="email"
                                      />
                                    </glamorous(input)>
                                    <div />
                                  </div>
                                </TextInput>
                              </TextInput>
                            </label>
                          </div>
                        </FormField>
                      </FormField>
                      <FormField
                        label="Password"
                        requiredText="Required"
                      >
                        <FormField
                          className={undefined}
                        >
                          <div
                            className="glamor-3"
                          >
                            <label>
                              <glamorous(div)
                                hideLabel={undefined}
                                key="formField-4-textWrapper"
                              >
                                <div
                                  className="glamor-0"
                                >
                                  <span
                                    id="formField-4-labelText"
                                  >
                                    Password
                                  </span>
                                </div>
                              </glamorous(div)>
                              <TextInput
                                aria-describedby={undefined}
                                disabled={true}
                                key="formField-4-control"
                                required={undefined}
                                rootProps={undefined}
                                type="password"
                                variant={undefined}
                              >
                                <TextInput
                                  className={undefined}
                                  variant={undefined}
                                >
                                  <div
                                    className="glamor-2"
                                  >
                                    <glamorous(input)
                                      aria-describedby={undefined}
                                      aria-invalid={undefined}
                                      aria-required={undefined}
                                      disabled={true}
                                      iconEnd={undefined}
                                      iconStart={undefined}
                                      innerRef={[Function]}
                                      prefix={undefined}
                                      readOnly={undefined}
                                      required={undefined}
                                      size="large"
                                      suffix={undefined}
                                      type="password"
                                      variant={undefined}
                                    >
                                      <input
                                        aria-describedby={undefined}
                                        aria-invalid={undefined}
                                        aria-required={undefined}
                                        className="glamor-1"
                                        disabled={true}
                                        prefix={undefined}
                                        readOnly={undefined}
                                        required={undefined}
                                        size="large"
                                        type="password"
                                      />
                                    </glamorous(input)>
                                    <div />
                                  </div>
                                </TextInput>
                              </TextInput>
                            </label>
                          </div>
                        </FormField>
                      </FormField>
                    </div>
                  </glamorous(div)>
                </fieldset>
              </FormFieldset>
            </FormFieldset>
          </div>
        </LivePreview>
      </div>
    </LiveProvider>
  </ThemeProvider>
</ThemeProvider>
`;

exports[`FormFieldset demo examples rtl 1`] = `
.glamor-4,
[data-glamor-4] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
}

.glamor-4 *,
[data-glamor-4] *,
.glamor-4 *::before,
[data-glamor-4] *::before,
.glamor-4 *::after,
[data-glamor-4] *::after {
  box-sizing: inherit;
}

.glamor-0,
[data-glamor-0] {
  color: #58606e;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  font-size: 0.875em;
  font-weight: 600;
  justify-content: space-between;
  margin-bottom: 0.5em;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
}

.glamor-0 > *,
[data-glamor-0] > * {
  align-self: flex-end;
  display: inline-block;
  -webkit-align-self: flex-end;
}

.glamor-3,
[data-glamor-3] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  align-items: center;
  cursor: text;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  position: relative;
  width: 100%;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-3 *,
[data-glamor-3] *,
.glamor-3 *::before,
[data-glamor-3] *::before,
.glamor-3 *::after,
[data-glamor-3] *::after {
  box-sizing: inherit;
}

.glamor-3 [role="img"],
[data-glamor-3] [role="img"] {
  display: block;
  fill: #c8d1e0;
  flex: 0 0 auto;
  margin: 0 0.5em;
  -webkit-flex: 0 0 auto;
}

.glamor-3 [role="img"]:last-of-type,
[data-glamor-3] [role="img"]:last-of-type {
  fill: #c8d1e0;
}

.glamor-5,
[data-glamor-5] {
  box-sizing: border-box;
  color: #333840;
  font-family: Open Sans, -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  font-size: 16px;
  line-height: 1.25;
  outline: 0;
  border: 1px solid transparent;
  border-top-color: #c8d1e0;
  padding: 0;
}

.glamor-5 *,
[data-glamor-5] *,
.glamor-5 *::before,
[data-glamor-5] *::before,
.glamor-5 *::after,
[data-glamor-5] *::after {
  box-sizing: inherit;
}

.glamor-5 > legend,
[data-glamor-5] > legend {
  color: #333840;
  font-size: 0.875em;
  font-weight: 700;
  line-height: 2em;
  padding: 0;
  padding-left: 0.5em;
}

.glamor-1,
[data-glamor-1] {
  fill: #8e99ab;
  font-size: 16px;
  height: 1.5em;
  transform: scaleX(-1);
  width: 1.5em;
  -webkit-transform: scaleX(-1);
}

.glamor-2,
[data-glamor-2] {
  background-color: transparent;
  border: 0;
  box-shadow: none;
  color: #58606e;
  flex: 1 1 auto;
  font-family: inherit;
  font-size: 0.875em;
  height: 2.857142857142857em;
  min-width: 0;
  outline: 0;
  padding-left: 1.1428571428571428em;
  padding-right: 0;
  width: 100%;
  -webkit-flex: 1 1 auto;
}

.glamor-2::placeholder,
[data-glamor-2]::placeholder,
.glamor-2[data-simulate-placeholder],
[data-glamor-2][data-simulate-placeholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-2::-ms-input-placeholder,
[data-glamor-2]::-ms-input-placeholder,
.glamor-2[data-simulate-msinputplaceholder],
[data-glamor-2][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-2:-ms-input-placeholder,
[data-glamor-2]:-ms-input-placeholder,
.glamor-2[data-simulate-msinputplaceholder],
[data-glamor-2][data-simulate-msinputplaceholder] {
  color: #8e99ab;
  font-style: italic;
}

.glamor-2::-ms-clear,
[data-glamor-2]::-ms-clear,
.glamor-2[data-simulate-msclear],
[data-glamor-2][data-simulate-msclear] {
  display: none;
}

.glamor-2:hover ~ div,
[data-glamor-2]:hover ~ div,
.glamor-2[data-simulate-hover] ~ div,
[data-glamor-2][data-simulate-hover] ~ div {
  border-color: #2e6fd9;
}

.glamor-2:focus ~ div,
[data-glamor-2]:focus ~ div,
.glamor-2[data-simulate-focus] ~ div,
[data-glamor-2][data-simulate-focus] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-2:active ~ div,
[data-glamor-2]:active ~ div,
.glamor-2[data-simulate-active] ~ div,
[data-glamor-2][data-simulate-active] ~ div {
  border-color: #c8d1e0;
  box-shadow: 0 0 0 1px #fff, 0 0 0 2px #2e6fd9;
}

.glamor-2 ~ div,
[data-glamor-2] ~ div {
  background-color: #fff;
  border-color: #c8d1e0;
  border-radius: 0.1875em;
  border-style: solid;
  border-width: 1px;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

<ThemeProvider>
  <ThemeProvider>
    <LiveProvider>
      <div
        className="react-live"
        scope={
          Object {
            "FormField": [Function],
            "FormFieldset": [Function],
            "IconBackspace": [Function],
            "TextInput": [Function],
            "ThemeProvider": [Function],
          }
        }
      >
        <LivePreview>
          <div
            className="react-live-preview"
          >
            <div
              dir="rtl"
            >
              <ThemeProvider>
                <ThemeProvider>
                  <FormFieldset
                    legend="أسطورة"
                  >
                    <FormFieldset>
                      <fieldset
                        className="glamor-5"
                      >
                        <legend>
                          أسطورة
                        </legend>
                        <FormField
                          label="مرحبا بالعالم"
                          requiredText="Required"
                        >
                          <FormField
                            className={undefined}
                          >
                            <div
                              className="glamor-4"
                            >
                              <label>
                                <glamorous(div)
                                  hideLabel={undefined}
                                  key="formField-5-textWrapper"
                                >
                                  <div
                                    className="glamor-0"
                                  >
                                    <span
                                      id="formField-5-labelText"
                                    >
                                      مرحبا بالعالم
                                    </span>
                                  </div>
                                </glamorous(div)>
                                <TextInput
                                  aria-describedby={undefined}
                                  defaultValue="مرحبا بالعالم"
                                  iconStart={<IconBackspace />}
                                  key="formField-5-control"
                                  required={undefined}
                                  rootProps={undefined}
                                  variant={undefined}
                                >
                                  <TextInput
                                    className={undefined}
                                    variant={undefined}
                                  >
                                    <div
                                      className="glamor-3"
                                    >
                                      <IconBackspace
                                        key="iconStart"
                                        size="1.5em"
                                      >
                                        <Icon
                                          rtl={true}
                                          size="1.5em"
                                        >
                                          <glamorous(svg)
                                            aria-hidden={true}
                                            aria-labelledby={undefined}
                                            role="img"
                                            rtl={true}
                                            size="1.5em"
                                            viewBox="0 0 24 24"
                                          >
                                            <svg
                                              aria-hidden={true}
                                              aria-labelledby={undefined}
                                              className="glamor-1"
                                              role="img"
                                              viewBox="0 0 24 24"
                                            >
                                              <g>
                                                <path
                                                  d="M22 3H7c-.69 0-1.23.35-1.59.88L0 12l5.41 8.11c.36.53.9.89 1.59.89h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-3 12.59L17.59 17 14 13.41 10.41 17 9 15.59 12.59 12 9 8.41 10.41 7 14 10.59 17.59 7 19 8.41 15.41 12 19 15.59z"
                                                />
                                              </g>
                                            </svg>
                                          </glamorous(svg)>
                                        </Icon>
                                      </IconBackspace>
                                      <glamorous(input)
                                        aria-describedby={undefined}
                                        aria-invalid={undefined}
                                        aria-required={undefined}
                                        defaultValue="مرحبا بالعالم"
                                        disabled={undefined}
                                        iconEnd={undefined}
                                        iconStart={<IconBackspace />}
                                        innerRef={[Function]}
                                        prefix={undefined}
                                        readOnly={undefined}
                                        required={undefined}
                                        size="large"
                                        suffix={undefined}
                                        type="text"
                                        variant={undefined}
                                      >
                                        <input
                                          aria-describedby={undefined}
                                          aria-invalid={undefined}
                                          aria-required={undefined}
                                          className="glamor-2"
                                          defaultValue="مرحبا بالعالم"
                                          disabled={undefined}
                                          prefix={undefined}
                                          readOnly={undefined}
                                          required={undefined}
                                          size="large"
                                          type="text"
                                        />
                                      </glamorous(input)>
                                      <div />
                                    </div>
                                  </TextInput>
                                </TextInput>
                              </label>
                            </div>
                          </FormField>
                        </FormField>
                      </fieldset>
                    </FormFieldset>
                  </FormFieldset>
                </ThemeProvider>
              </ThemeProvider>
            </div>
          </div>
        </LivePreview>
      </div>
    </LiveProvider>
  </ThemeProvider>
</ThemeProvider>
`;
